edgedb/retry.py

```diff
@@ -0,0 +1,163 @@
+import asyncio
+import random
+import time
+
+from . import errors
+from . import transaction as _transaction
+
+
+DEFAULT_MAX_ITERATIONS = 3
```

Comment:
We need to make this a configurable option for connect() etc.

Reply:
Well. We discussed this with @elprans and he suggested making it non-configuable for now.

The issue is:
1. RFC1004 provisions `with_retry_options` for this.
2. We don't want to add more options to `connect()` because we want to make a builder/fluent interface for that
3. Even if we want to add a temporary option to `connect()`, retry options is more complex than that, you're supposed to configure different retry options for different conditions (i.e. on deadlock and network error) and also retry interval (function).
4. Even if we want to make builder/fluent now or eventually, it makes little sense to add retry count to that initial connection options, because overriding rules, are not clear. For example, whether `conn_builder.retry_options(DEADLOCK, 2)` overrides `pool.with_retry_options(ALL, 3)`.

So as I've already said: RFC1004 provides a nice coherent way to configure transaction options. If you want to remove that part of it, it should be better specified how to configure all of that.

Reply:
> RFC1004 provisions with_retry_options for this.

FWIW I realized a few weeks ago that the only argument I had against that idea doesn't hold when we have a server side pool of connections. And we do now. So I withdraw my objections to the proposed `with_*` design.

> Well. We discussed this with @elprans and he suggested making it non-configuable for now.

Yeah, everything you said sounds reasonable. Let's merge as is and we'll later make it configurable via with_ or some other well-thought API.